### PR TITLE
Second step of jbuilder migration

### PR DESCRIPTION
**Pull request by ghost**
The `jbuilder` binary now emits a warning on every startup and both `jbuilder` and `dune` emits warning when encountering `jbuild` files.

### CODE REVIEW

**Review by rgrinberg — APPROVED** [review]
Looks good. Perhaps we should merge `$ dune upgrade` first and update many of these tests? That would minimize this diff by quite a bit.

**Comment by diml** [comment]
Make sense

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 unreleased
 ----------
 
+- Second step of the deprecation of jbuilder: the `jbuilder` binary
+  now emits a warning on every startup and both `jbuilder` and `dune`
+  emits warning when encountering `jbuild` files (#...., @diml)
```
Comment (emillon) [inline CHANGES.md]:
```suggestion
  emit warnings when encountering `jbuild` files (#1752, @diml)
```

File: src/file_tree.ml [hunk 2]
```diff
@@ -177,10 +177,17 @@ let load path =
             match List.filter ["dune"; "jbuild"] ~f:(String.Set.mem files) with
             | [] -> (None, Sub_dirs.default)
             | [fn] ->
+              let file = Path.relative path fn in
               if fn = "dune" then
-                Dune_project.ensure_project_file_exists project;
+                Dune_project.ensure_project_file_exists project
+              else
+                Errors.warn (Loc.in_file file)
+                  "jbuild files are depreacted, please convert this file to \
```
Comment (emillon) [inline src/file_tree.ml]:
```suggestion
                  "jbuild files are deprecated, please convert this file to \
```

**Comment by diml** [comment]
I did find an issue with the upgrader while trying to upgrade the tests, I'll fix that first

**Comment by diml** [comment]
I rebased this PR and did another pass on the testsuite. I replaced a few calls to `jbuilder` by `dune` wherever appropriate.

**Review by rgrinberg — APPROVED** [review]

**Pull request merged**